eventmesh-registry-plugin/eventmesh-registry-etcd/src/main/java/org/apache/eventmesh/registry/etcd/service/EtcdCustomService.java

```diff
@@ -77,31 +77,29 @@ public List<EventMeshServicePubTopicInfo> findEventMeshServicePubTopicInfos() th
 
     @Nullable
     public EventMeshAppSubTopicInfo findEventMeshAppSubTopicInfoByGroup(String group) throws RegistryException {
-
         Client client = getEtcdClient();
         String keyPrefix = KEY_PREFIX + KEY_APP + EtcdConstant.KEY_SEPARATOR + group;
         List<KeyValue> keyValues = null;
         try {
             ByteSequence keyByteSequence = ByteSequence.from(keyPrefix.getBytes(Constants.DEFAULT_CHARSET));
-
             GetOption getOption = GetOption.newBuilder().withPrefix(keyByteSequence).build();
-
             keyValues = client.getKVClient().get(keyByteSequence, getOption).get().getKvs();
-
-
             if (CollectionUtils.isNotEmpty(keyValues)) {
-                for (KeyValue kv : keyValues) {
-                    EventMeshAppSubTopicInfo eventMeshAppSubTopicInfo =
-                        JsonUtils.parseObject(new String(kv.getValue().getBytes(), Constants.DEFAULT_CHARSET), EventMeshAppSubTopicInfo.class);
+                EventMeshAppSubTopicInfo eventMeshAppSubTopicInfo =
+                    JsonUtils.parseObject(
+                        new String(keyValues.get(0).getValue().getBytes(), Constants.DEFAULT_CHARSET),
+                        EventMeshAppSubTopicInfo.class
+                    );
+                if (eventMeshAppSubTopicInfo != null) {
                     return eventMeshAppSubTopicInfo;
+                } else {
+                    throw new NullPointerException("eventMeshAppSubTopicInfo is null!");
```

Comment:
Return directly, there is already a null check outside.

Reply:
Done